.buildkite/generate_pipeline.py

```diff
@@ -0,0 +1,254 @@
+"""
+This script generates a Buildkite pipeline from test files.
+
+The script will generate two pipelines:
+
+tests/smoke_tests
+├── test_*.py -> release pipeline
+├── test_pre_merge.py -> pre-merge pipeline
+
+run `python .buildkite/generate_pipeline.py` to generate the pipeline for
+testing. The CI will run this script as a pre-step, and use the generated
+pipeline to run the tests.
+
+1. release pipeline, which runs all smoke tests by default, generates all
+   smoke tests for all clouds.
+2. pre-merge pipeline, which generates all smoke tests for all clouds,
+   author should specify which clouds to run by setting env in the step.
+
+We only have credentials for aws/azure/gcp/kubernetes(CLOUD_QUEUE_MAP and
+SERVE_CLOUD_QUEUE_MAP) now, smoke tests for those clouds are generated, other
+clouds are not supported yet, smoke tests for those clouds are not generated.
+"""
+
+import ast
+import os
+import random
+from typing import Any, Dict, List, Optional
+
+import yaml
+
+DEFAULT_CLOUDS_TO_RUN = ['aws', 'azure']
+
+ALL_CLOUDS_IN_SMOKE_TESTS = [
```

Comment:
Is it possible to get this from conftest.py to avoid having to keep them in sync? Is it not possible to import that file from here?

Reply:
Yes, but pytest has different import levels. The code could look more ugly.
Changed.